tidy3d/components/data/dataset.py

```diff
@@ -955,6 +1019,131 @@ def interp(
         x = np.atleast_1d(x)
         y = np.atleast_1d(y)
         z = np.atleast_1d(z)
+
+        if method == "nearest":
+            interpolated_values = self._interp_nearest(x=x, y=y, z=z)
+        else:
+            if fill_value == "extrapolate" and method != "nearest":
```

Comment:
don't need `method != nearest` in this check or below, since we are in `else`

Reply:
fixed